jerry-core/parser/js/js-parser.c

```diff
@@ -2302,6 +2302,12 @@ parser_parse_function (parser_context_t *context_p, /**< context */
   }
 
   lexer_next_token (context_p);
+#if ENABLED (JERRY_ESNEXT)
+  if ((context_p->status_flags & (PARSER_CLASS_CONSTRUCTOR | PARSER_ALLOW_SUPER_CALL)) == PARSER_CLASS_CONSTRUCTOR)
+  {
+    parser_flush_cbc (context_p);
+  }
+#endif /* ENABLED (JERRY_ESNEXT) */
```

Comment:
There is a code above:
```
if ((context_p->status_flags & (PARSER_CLASS_CONSTRUCTOR | PARSER_ALLOW_SUPER_CALL)) == PARSER_CLASS_CONSTRUCTOR)
  {
    parser_emit_cbc_ext (context_p, CBC_EXT_RUN_FIELD_INIT);
  }
```
Maybe we could put a flush after it. But what cause the flush in a normal case (when no parse-error)?

Reply:
The linked issue also occurs with correct syntax, any time a constructor begins with a function statement. No parse error happens when I am stepping through the example with `cgdb`. For example this fails:
```js
class foo  {
    constructor() {
        function bar() {}
    }
}
```
But this runs as expected:
```js
class foo  {
    constructor() {
        var a = 5
        function bar() {}
    }
}
```
In the second case, a flush occurs in `parser_emit_cbc_literal()`, that's why the `context_p->last_cbc_opcode == PARSER_CBC_UNAVAILABLE` check will pass when we begin parsing the `function bar() {}`. I didn't check for every kind of statement, but I think there must be some kind of flush in all of them.
And the tests are all passing so unless there is some corner case we don't test, I think this should have no side effects.

Reply:
I also asked: what about putting the flush after the CBC_EXT_RUN_FIELD_INIT?

Reply:
Yes, I forgot about that, sorry. We can put it there of course, I will update the patch.